lib/connectors/AsyncSentinelConnector.ts

```diff
@@ -0,0 +1,35 @@
+import SentinelConnector, {
+    ISentinelConnectionOptions,
+    ISentinelConnectorOptions,
+    ISentinelAddress,
+    SentinelIterator,
+    EMPTY_SENTINELS_MSG } from './SentinelConnector';
+import { ErrorEmitter } from './AbstractConnector';
+import { NetStream } from '../types';
+import { NodeCallback } from './types';
+
+export type AsyncSentinelFetch = () => Promise<Partial<ISentinelAddress>[]>
+
+export default class AsyncSentinelConnector extends SentinelConnector {
+    private fetch: AsyncSentinelFetch
+
+    constructor(options: ISentinelConnectionOptions, fetch: AsyncSentinelFetch) {
+        options.sentinels = options.sentinels || [{host: 'localhost', port: 6379}] // Placeholder
+        super(options as ISentinelConnectorOptions)
+
+        this.fetch = fetch;
+    }
+
+    public connect(callback: NodeCallback<NetStream>, eventEmitter: ErrorEmitter) {
```

Comment:
I prefer to making all public APIs use Promise instead of the outdated callback-style. So if we are going to export the connectors, the `connect` method needs to be changed to `public connect(eventEmitter: ErrorEmitter): Promise<NetStream>`.